src/modules/admin/Admin.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import AdminTabCategory from './models';
+import AdminTabCategories from './admin-tab-categories/AdminTabCategories';
+import TechnologiesTab from './technologies-tab/TechnologiesTab';
+import PatternsTab from './patterns-tab/PatternsTab';
+
+import csx from './Admin.scss';
+
+const renderTabContent = (category: string) => {
```

Comment:
you can save a lot of code with Strategy Design Pattern based on object literal:

const tabs = {
   [AdminTabCategory.TECHNOLOGIES]: TechnologiesTab,
};

replace tabContent with tabs[category]()

remove use MEmo and remove totally method for rendering purposes